graphein/ml/datasets/torch_geometric_dataset.py

```diff
@@ -130,54 +130,56 @@ def __init__(
         self.pdb_codes = (
             [pdb.lower() for pdb in pdb_codes]
             if pdb_codes is not None
-            else None
+            else []
         )
         self.uniprot_ids = (
             [up.upper() for up in uniprot_ids]
             if uniprot_ids is not None
-            else None
+            else []
         )
-
         self.pdb_paths = pdb_paths
-        if self.pdb_paths is None:
-            if self.pdb_codes and self.uniprot_ids:
-                self.structures = self.pdb_codes + self.uniprot_ids
-            elif self.pdb_codes:
-                self.structures = pdb_codes
-            elif self.uniprot_ids:
-                self.structures = uniprot_ids
-        # Use local saved pdb_files instead of download or move them to self.root/raw dir
-        else:
-            if isinstance(self.pdb_paths, list):
-                self.structures = [
+        #  make sure root path is unique
+        if self.pdb_paths:
+            #  add pdb_paths' name into self.structure
+            self.pdb_paths_name = [
                     os.path.splitext(os.path.split(pdb_path)[-1])[0]
                     for pdb_path in self.pdb_paths
                 ]
-                self.pdb_path, _ = os.path.split(self.pdb_paths[0])
-
-        if self.pdb_codes and self.uniprot_ids:
-            self.structures = self.pdb_codes + self.uniprot_ids
-        elif self.pdb_codes:
-            self.structures = pdb_codes
-        elif self.uniprot_ids:
-            self.structures = uniprot_ids
-        self.af_version = af_version
+            #  if root pdb_path is not unique raise error since we will save all pdb into this root pdb_path and take it as the self.raw_dir
+            if len(set([os.path.split(pdb_path)[0] for pdb_path in pdb_paths])) != 1:
+                raise ValueError("pdb_paths should have only one root path not so much!")
+        else:
+            self.pdb_paths_name = []
+
+        self.structures = list(set(self.pdb_codes + self.uniprot_ids + self.pdb_paths_name))  # remove some pdb_codes is in pdb_path and loaded repeately
```

Comment:
I don't think this should be a set operation. With chain selections you may want to have e.g. `3eiy_A` and `3eiy_B` as different examples in your dataset.

Reply:
Well, i guess it would'n make a difference at chain selection, this set operation is to drop duplicate in the result list of `pdb_codes` + `uniprot_ids` + `paths_name`. As you can see, local dir may have some pdb files like `10gs.pdb`, and if `pdb_codes` also have `10gs` to download,  and self.structures would contain double `10gs` and so the finial `dataset` object will have duplicate `Data` object.

Reply:
It becomes a problem here though (L283), no?

```python
    def process(self):
        """Process structures into PyG format and save to disk."""
        # Read data into huge `Data` list.
        structure_files = [
            f"{self.raw_dir}/{pdb}.pdb" for pdb in self.structures
        ]
```

Reply:
Well, i guess not.
code like below in the `tests/ml/test.ipynb`
```python
from graphein.ml.datasets import InMemoryProteinGraphDataset

local_dir = "../protein/test_data"
pdb_paths = [osp.join(local_dir, pdb_file) for pdb_file in os.listdir(local_dir) if pdb_file.endswith(".pdb")]

ds = InMemoryProteinGraphDataset(root = "../protein/test_data/InMemoryProteinGraphDataset",
                    name = "InMemoryProteinGraphDataset_test",
                    pdb_paths=pdb_paths,
                    pdb_codes=["10gs"],
                    uniprot_ids=["A0A6J1BG53", "A0A6P5Z5F7"],
                    af_version=3)

```

and before running it:
<img width="233" alt="image" src="https://user-images.githubusercontent.com/49584439/191213001-30b9ffbe-8d86-4db2-a5ac-765f3779844f.png">

then run it :
<img width="852" alt="image" src="https://user-images.githubusercontent.com/49584439/191213682-995ebb51-c38e-4795-98c6-d31102e7177f.png">

Reply:
Could you see what happens with:


```python
from graphein.ml.dataset import InMemoryProteinGraphDataset

ds = InMemoryProteinGraphDataset(root = ""../protein.test_data/InMemoryProteinGraphDataset", pdb_paths=pdb_paths, pdb_codes = ["4hhb", "4hhb"], chain_selection=["A","B"])

```

Reply:
well, i'll try later

Reply:
> Could you see what happens with:
> 
> ```python
> from graphein.ml.dataset import InMemoryProteinGraphDataset
> 
> ds = InMemoryProteinGraphDataset(root = ""../protein.test_data/InMemoryProteinGraphDataset", pdb_paths=pdb_paths, pdb_codes = ["4hhb", "4hhb"], chain_selection=["A","B"])
> ```
<img width="1285" alt="image" src="https://user-images.githubusercontent.com/49584439/191711350-7702a5c5-f73d-40c1-a473-be4774e4aeea.png">

and why i `['4hhs', '4hhs']` 

<img width="648" alt="image" src="https://user-images.githubusercontent.com/49584439/191711529-f7ae87f0-8cc5-4320-8fa3-cf63ff920cba.png">

I guess this may need lots of change~